File: src/main/java/com/badahori/creatures/plugins/intellij/agenteering/caos/stubs/types/CaosScriptStubTypes.java
package com.badahori.creatures.plugins.intellij.agenteering.caos.stubs.types;

import org.jetbrains.annotations.NotNull;

public interface CaosScriptStubTypes {
    @NotNull
    CaosScriptCommandCallStubType COMMAND_CALL = new CaosScriptCommandCallStubType("CaosScript_COMMAND_CALL");

    @NotNull
    CaosScriptSubroutineStubType SUBROUTINE = new CaosScriptSubroutineStubType("CaosScript_SUBROUTINE");

    @NotNull
    CaosScriptVarTokenStubType VAR_TOKEN = new CaosScriptVarTokenStubType("CaosScript_VAR_TOKEN");

    @NotNull
    CaosScriptTargAssignmentStubType TARG_ASSIGNMENT = new CaosScriptTargAssignmentStubType("CaosScript_CTarg");

    @NotNull
    CaosScriptLValueStubType LVALUE = new CaosScriptLValueStubType("CaosScript_LVALUE");

    @NotNull
    CaosScriptRValueStubType RVALUE = new CaosScriptRValueStubType("CaosScript_RVALUE");

    @NotNull
    CaosScriptTokenRValueStubType TOKEN_RVALUE = new CaosScriptTokenRValueStubType("CaosScript_TOKEN_RVALUE");

    @NotNull
    CaosScriptAssignmentStubType VAR_ASSIGNMENT = new CaosScriptAssignmentStubType("CaosScript_CAssignment");

    @NotNull
    CaosScriptNamedGameVarStubType NAMED_GAME_VAR = new CaosScriptNamedGameVarStubType("CaosScript_NAMED_GAME_VAR");

    @NotNull
    CaosScriptEventScriptStubType EVENT_SCRIPT = new CaosScriptEventScriptStubType("CaosScript_EVENT_SCRIPT");

    @NotNull
    CaosScriptRndvStubType RNDV = new CaosScriptRndvStubType("CaosScript_C_RNDV");


    @NotNull
    CaosScriptMacroStubType MACRO = new CaosScriptMacroStubType("CaosScript_MACRO");

    @NotNull
    CaosScriptInstallScriptStubType INSTALL_SCRIPT = new CaosScriptInstallScriptStubType("CaosScript_INSTALL_SCRIPT");

    @NotNull
    CaosScriptRemovalScriptStubType REMOVAL_SCRIPT = new CaosScriptRemovalScriptStubType("CaosScript_REMOVAL_SCRIPT");

    @NotNull
    CaosScriptFileStubType FILE = new CaosScriptFileStubType();

    @NotNull
    CaosScriptRValuePrimeStubType RVALUE_PRIME = new CaosScriptRValuePrimeStubType("CaosScript_RVALUE_PRIME");
}

File: src/main/java/com/badahori/creatures/plugins/intellij/agenteering/caos/stubs/types/CaosScriptStubTypes.kt
package com.badahori.creatures.plugins.intellij.agenteering.caos.stubs.types

interface CaosScriptStubTypes {
    companion object {
        @JvmStatic
        val COMMAND_CALL = CaosScriptCommandCallStubType("CaosScript_COMMAND_CALL")
        @JvmStatic
        val CAOS_2_BLOCK = CaosScriptCaos2BlockStubType("CaosScript_CAOS_2_BLOCK")
        @JvmStatic
        val CAOS_2_TAG = CaosScriptCaos2TagStubType("CaosScript_CAOS_2_TAG")
        @JvmStatic
        val CAOS_2_COMMAND = CaosScriptCaos2CommandStubType("CaosScript_CAOS_2_COMMAND")
        @JvmStatic
        val SUBROUTINE = CaosScriptSubroutineStubType("CaosScript_SUBROUTINE")
        @JvmStatic
        val VAR_TOKEN = CaosScriptVarTokenStubType("CaosScript_VAR_TOKEN")
        @JvmStatic
        val TARG_ASSIGNMENT = CaosScriptTargAssignmentStubType("CaosScript_CTarg")
        @JvmStatic
        val LVALUE = CaosScriptLValueStubType("CaosScript_LVALUE")
        @JvmStatic
        val RVALUE = CaosScriptRValueStubType("CaosScript_RVALUE")
        @JvmStatic
        val TOKEN_RVALUE = CaosScriptTokenRValueStubType("CaosScript_TOKEN_RVALUE")
        @JvmStatic
        val VAR_ASSIGNMENT = CaosScriptAssignmentStubType("CaosScript_CAssignment")
        @JvmStatic
        val NAMED_GAME_VAR = CaosScriptNamedGameVarStubType("CaosScript_NAMED_GAME_VAR")
        @JvmStatic
        val EVENT_SCRIPT = CaosScriptEventScriptStubType("CaosScript_EVENT_SCRIPT")
        @JvmStatic
        val RNDV = CaosScriptRndvStubType("CaosScript_C_RNDV")
        @JvmStatic
        val MACRO = CaosScriptMacroStubType("CaosScript_MACRO")
        @JvmStatic
        val INSTALL_SCRIPT = CaosScriptInstallScriptStubType("CaosScript_INSTALL_SCRIPT")
        @JvmStatic
        val REMOVAL_SCRIPT = CaosScriptRemovalScriptStubType("CaosScript_REMOVAL_SCRIPT")
        @JvmStatic
        val SUBROUTINE_NAME = CaosScriptSubroutineNameStubType("CaosScript_SUBROUTINE_NAME")
        @JvmStatic
        val FILE = CaosScriptFileStubType()
        @JvmStatic
        val RVALUE_PRIME = CaosScriptRValuePrimeStubType("CaosScript_RVALUE_PRIME")
    }
}